src/main/webapp/app/text-assessment/text-assessment.component.ts

```diff
@@ -308,6 +314,14 @@ export class TextAssessmentComponent implements OnInit, OnDestroy, AfterViewInit
             this.generalFeedback = new Feedback();
         }
         this.referencedFeedback = feedbacks;
+        this.referencedTextBlocks = feedbacks.map(feedback => {
```

Comment:
I think this is ok, but it could be improved from a readability view point.

As a suggestion:

```
import { intersectionWith, zip } from lodash;
...

const feedbackReferencesTextBlocks = feedbacks.filter(({reference}) => this.sha1Regex.test(feedback.reference!));
this.referencedTextBlocks = intersectionWith(this.submission.blocks, feedbackReferencesTextBlocks, (block, feedback) => block.id === fedback.reference);
```

At least I like it better this way :sunglasses: 

Also it would be nice if we could avoid some of the null check workarounds.

Reply:
TBH, I am not sure if this necessarily readability for all developers. I do like the `intersectionWith` approach, however this will break the assumption state above:
> Yes, we guarantee equal length for `referencedFeedback` and `referencedTextBlocks`.

This should be retouched when reworking the Data Structure.